Source/Peripheral.swift

```diff
@@ -148,11 +148,10 @@ public class Peripheral {
         }
         .take(1)
 
-        return .create { observer in
-            let disposable = self.ensureValidPeripheralState(for: observable).subscribe(observer)
-            self.peripheral.discoverServices(serviceUUIDs)
-            return disposable
-        }
+        return ensureValidStateAndCallIfNecessary(
+            for: observable,
+            call: { self.peripheral.discoverServices(serviceUUIDs) }
```

Comment:
I'm not sure how to point out that this call is always called _after_ observable subscription.